src/alchemlyb/workflows/abfe.py

```diff
@@ -0,0 +1,665 @@
+            self.file_list))
+
+        if software.lower() == 'gromacs':
+            self.logger.info(f'Using {software} parser to read the data.')
+            self._extract_u_nk = gmx.extract_u_nk
+            self._extract_dHdl = gmx.extract_dHdl
+        elif software.lower() == 'amber':
+            self._extract_u_nk = amber.extract_u_nk
+            self._extract_dHdl = amber.extract_dHdl
+        else: # pragma: no cover
+            raise NameError(f'{software} parser not found.')
+
+    def read(self):
+        '''Read the u_nk and dHdL data from the
+        :attr:`~alchemlyb.workflows.ABFE.file_list`
+
+        Attributes
+        ----------
+        u_nk_list : list
+            A list of :class:`pandas.DataFrame` of u_nk.
+        dHdl_list : list
+            A list of :class:`pandas.DataFrame` of dHdl.
+        '''
+        u_nk_list = []
+        dHdl_list = []
+        for xvg in self.file_list:
+            try:
+                u_nk = self._extract_u_nk(xvg, T=self.T)
+                self.logger.info(
+                    f'Reading {len(u_nk)} lines of u_nk from {xvg}')
+                u_nk_list.append(u_nk)
+            except: # pragma: no cover
+                self.logger.warning(
+                    f'Error reading read u_nk from {xvg}.')
+
+            try:
+                dhdl = self._extract_dHdl(xvg, T=self.T)
+                self.logger.info(
+                    f'Reading {len(dhdl)} lines of dhdl from {xvg}')
+                dHdl_list.append(dhdl)
+            except: # pragma: no cover
+                self.logger.warning(
+                    f'Error reading read dhdl from {xvg}.')
+
+        # Sort the files according to the state
+        if len(u_nk_list) > 0:
+            self.logger.info('Sort files according to the u_nk.')
+            column_names = u_nk_list[0].columns.values.tolist()
+            index_list = sorted(range(len(self.file_list)),
+                key=lambda x:column_names.index(
+                    u_nk_list[x].reset_index('time').index.values[0]))
+        else:
+            self.logger.info('Sort files according to the dHdl.')
+            column_names = sorted([dHdl.reset_index('time').index.values[0]
+                                   for dHdl in dHdl_list])
+            index_list = sorted(range(len(self.file_list)),
+                key=lambda x:column_names.index(
+                    dHdl_list[x].reset_index('time').index.values[0]))
+
+        self.file_list = [self.file_list[i] for i in index_list]
+        self.logger.info("Sorted file list: \n %s", '\n'.join(self.file_list))
+        self.u_nk_list = [u_nk_list[i] for i in index_list]
+        self.dHdl_list = [dHdl_list[i] for i in index_list]
+        self.u_nk_sample_list = None
+        self.dHdl_sample_list = None
+
+    def run(self, skiptime=0, uncorr='dhdl', threshold=50,
+            methods=('mbar', 'bar', 'ti'), overlap='O_MBAR.pdf',
+            breakdown=True, forwrev=10, *args, **kwargs):
```

Comment:
Given that you have kwargs with defaults, you might not need to specify the defaults in the text. That makes it easier to keep everything consistent.

Reply:
@orbeckst Sorry, I wonder what do you mean by specify the defaults in the text. What is this test?

Reply:
In the docstring (=text) you often state the default for a kwarg. There's nothing wrong with doing that and you don't really need to change anything. I was only saying that *if* the defaults are set with the kwargs themselves then sphinx will generate documentation where the defaults are visible in the function signature and then it's not strictly necessary to manually write the defaults in the doc string. That has the advantage of not having to update the default in the docstring if the actual default value in the code ever gets changed.

It's a minor thing and as I said, you don't really need to change your docstring. If you _want_ to omit manually written defaults in the future then I'd be happy with that. That's all I really wanted to say.